service/worker/esanalyzer/workflow.go

```diff
@@ -171,6 +216,123 @@ func (w *Workflow) emitWorkflowVersionMetrics(ctx context.Context) error {
 	return nil
 }
 
+func (w *Workflow) emitWorkflowVersionMetricsPinot(domainName string, logger *zap.Logger) error {
+	wfVersionPinotQuery, err := w.getWorkflowTypePinotQuery(domainName)
+	if err != nil {
+		logger.Error("Failed to get Pinot query to find workflow version Info",
+			zap.Error(err),
+			zap.String("DomainName", domainName),
+		)
+		return err
+	}
+	response, err := w.analyzer.pinotClient.SearchAggr(&pinot.SearchRequest{Query: wfVersionPinotQuery})
+	if err != nil {
+		logger.Error("Failed to query Pinot to find workflow type count Info",
+			zap.Error(err),
+			zap.String("VisibilityQuery", wfVersionPinotQuery),
+			zap.String("DomainName", domainName),
+		)
+		return err
+	}
+	foundAggregation := len(response) > 0
+
+	if !foundAggregation {
+		logger.Error("Pinot error: aggregation failed.",
+			zap.Error(err),
+			zap.String("Aggregation", fmt.Sprintf("%v", response)),
+			zap.String("DomainName", domainName),
+			zap.String("VisibilityQuery", wfVersionPinotQuery),
+		)
+		return err
```

Comment:
this will return nil because aggregation is empty

Reply:
I noticed that ES has the same issue. We didn't catch that previously

Reply:
handled for both Pinot and ES